lean4-lake.el

```diff
@@ -30,13 +30,13 @@
 (defun lean4-lake-find-dir-in (dir)
   "Find a parent directory of DIR with file \"lakefile.lean\"."
   (when dir
-    (or (when (f-exists? (f-join dir "lakefile.lean")) dir)
-	(lean4-lake-find-dir-in (f-parent dir)))))
+    (or (when (file-exists-p (expand-file-name "lakefile.lean" dir)) dir)
+	(lean4-lake-find-dir-in (file-name-directory (directory-file-name dir))))))
 
 (defun lean4-lake-find-dir ()
   "Find a parent directory of the current file with file \"lakefile.lean\"."
   (and (buffer-file-name)
-       (lean4-lake-find-dir-in (f-dirname (buffer-file-name)))))
+       (lean4-lake-find-dir-in (directory-file-name (buffer-file-name)))))
```

Comment:
You can use `locate-dominating-file` instead of the `lean4-lake-find-dir-in` function defined above:

```emacs-lisp
(locate-dominating-file default-directory "lakefile.lean")
```